pkg/kubectl/cmd/util/shortcut_restmapper_test.go

```diff
@@ -194,3 +200,76 @@ func TestKindFor(t *testing.T) {
 		}
 	}
 }
+
+type fakeDiscoveryClient struct {
+	groupCalls    int
```

Comment:
slim down to what is actually needed.

@juanvallejo someone was working on a real fake discovery client.  We need to find that pull again.

Reply:
@deads2k 
> someone was working on a real fake discovery client. We need to find that pull again.

was it https://github.com/kubernetes/kubernetes/pull/62069 ?